src/python/pants/invalidation/build_invalidator.py

```diff
@@ -58,7 +58,9 @@ def __init__(self, cache_key_gen_version=None):
     cache_key_gen_version - If provided, added to all cache keys. Allows you to invalidate
       all cache keys in a single pants repo, by changing this value in config.
     """
-
+    # TODO: Qualify the BuildInvalidator root with the cache_key_gen_version, instead of mixing
+    # it into the key, for uniformity with other qualifiers like the fprint strategy and the
```

Comment:
Right, but doesn't mixing it into the path have the same result as mixing it into the contents of the file? If you bump the cache_key_gen_version the file will be missing from the new path, so everything will get invalidated. 

I guess if you then demote the cache_key_gen_version back to its old value you can get into an inconsistent state, because Pants will find the old file and assume that nothing needs rebuilding, even though the workdir reflects the state of the new cache_key_gen_version.